src/redux/books/booksSlice.js

```diff
@@ -1,42 +1,70 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { v4 as uuidv4 } from 'uuid';
 
-const initialState = {
-  allBooks: [
-    {
-      item_id: 'item1',
-      title: 'The Great Gatsby',
-      author: 'John Smith',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item2',
-      title: 'Anna Karenina',
-      author: 'Leo Tolstoy',
-      category: 'Fiction',
+// unique api url
+const api = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/eHu0AJvEHeHUitjoncwV/books';
+
+// fetching data from api
+export const fetchBooksData = createAsyncThunk('fetchBooksData', async () => {
+  const response = await fetch(api);
```

Comment:
- [ ] As required by project requirements, I kindly suggest that all HTTP requests are mad with `axios`.

Reply:
Hello @topeogunleye 
I hope you are doing well. As mentioned in the requirements, we have a choice to use in fetch and axios.

Reply:
Sorry, my bad, please kindly submit again and you'll be approved

Reply:
Thank you so much @topeogunleye

Reply:
@MAleemHm, you're welcome